### PR TITLE
feat: add named prom port to dataplane sidecar

### PR DESCRIPTION
**Pull request by DanStough**
Changes proposed in this PR:
- This port of #1440 is updated for consul-dataplane and the v2 API. It adds a named port, `prometheus` to the container. This can be used for the Prometheus operator.

This could easily be backported to previous versions, but I figured there was no point in the risk, considering how long this issue has been around. Let me know if anyone disagrees. I will need to do it manually to remove the V2 changes.

How I've tested this PR: unit tests

How I expect reviewers to test this PR: 👀 

Checklist:
- [X] Tests added
- [X] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Comment by david-yu** [comment]
@DanStough I would backport to at least 1.3.0 since right now there is no option for folks to use this feature and they will have to wait until v2 is GA.

File: control-plane/connect-inject/webhook/consul_dataplane_sidecar.go [hunk 1]
```diff
@@ -162,6 +162,15 @@ func (w *MeshWebhook) consulDataplaneSidecar(namespace corev1.Namespace, pod cor
 		container.VolumeMounts = append(container.VolumeMounts, volumeMounts...)
 	}
 
+	// Container Ports
+	ports, err := w.getContainerPorts(pod)
+	if err != nil {
+		return corev1.Container{}, err
+	}
+	if ports != nil {
+		container.Ports = ports
```
Comment (curtbushko) [inline control-plane/connect-inject/webhook/consul_dataplane_sidecar.go:171]: Doesn't this incorrectly wipe out all the ports already set? Should it be an append?

On line 149:

		container.Ports = append(container.Ports, corev1.ContainerPort{
			Name:          fmt.Sprintf("%s-%d", "proxy-health", mpi.serviceIndex),
			ContainerPort: int32(constants.ProxyDefaultHealthPort + mpi.serviceIndex),

**Review by curtbushko — CHANGES_REQUESTED** [review]
Thanks Dan!  Looks pretty good but I think that I may? have found a bug?

File: control-plane/connect-inject/webhook/consul_dataplane_sidecar.go (hunk 1 above)
Comment (DanStough) [reply control-plane/connect-inject/webhook/consul_dataplane_sidecar.go:171]: Great catch! 🙇

**Review by curtbushko — APPROVED** [review]
LGTM!

**Pull request merged**